arkouda/segarray.py

```diff
@@ -91,88 +90,80 @@ def update_doc():
 
 def segarray(segments: pdarray, values: pdarray, lengths=None, grouping=None):
     """
-    Alias for the from_parts function. Prevents user from needing to call `ak.SegArray.from_parts`
+    Alias for the from_parts function. Prevents user from needing to call `ak.SegArray` constructor
+    DEPRECATED
     """
-    return SegArray.from_parts(segments, values, lengths, grouping)
+    warn(
+        "ak.segarray has been deprecated. Please use ak.SegArray constructor moving forward",
+        DeprecationWarning,
+    )
+    return SegArray(segments, values, lengths, grouping)
 
 
 class SegArray:
     objType = "SegArray"
 
-    def __init__(
-        self, name, dtype, size, ndim, shape, itemsize, segments, values, lengths=None, grouping=None
-    ):
-        self.name = name
-        self.dtype = dtype
-        self.size = size
-        self.ndim = ndim
-        self.shape = shape
-        self.itemsize = itemsize
-
+    def __init__(self, segments, values, lengths=None, grouping=None):
         self.logger = getArkoudaLogger(name=__class__.__name__)  # type: ignore
+        self.name = ""
+
+        # validate inputs
+        if not isinstance(segments, pdarray) or segments.dtype != akint64:
+            raise TypeError("Segments must be int64 pdarray")
+        if not isinstance(values, pdarray):
+            raise TypeError("Values must be a pdarray.")
+        if not is_sorted(segments):
+            raise ValueError("Segments must be unique and in sorted order")
+        if segments.size > 0:
+            if segments.min() != 0:
+                raise ValueError("Segments must start at zero and be less than values.size")
```

Comment:
Yeah it could. I had just moved the block over from `from_parts`. The message also needs to be updated now that I reread it because we allow for empty segments.

Reply:
Updated the if and combined with the outer layer instead of nesting.I updated the error message as well